lewis/core/simulation.py

```diff
@@ -110,6 +112,38 @@ def _create_control_server(self, control_server):
                                         exclude=('add_adapter', 'remove_adapter', 'handle'))},
             control_server)
 
+    @property
+    def setups(self):
+        """
+        A list of setups that are available. Use :meth:`switch_setup` to
+        change the setup.
+        """
+        return list(self._device_builder.setups.keys()) if self._device_builder is not None else []
+
+    def switch_setup(self, new_setup):
+        """
+        This method switches the setup, which means that it replaces the currently
+        simulated device with a new device, as defined by the setup. This resets
+        the timers that keep track of the simulation.
+
+        If any error occurs during setup switching it is logged and re-raised.
+
+        :param new_setup: Name of the new setup to load.
+        """
+        try:
+            self._device = self._device_builder.create_device(new_setup)
+            self._reset_timers()
+            self._adapters.device = self._device
+        except Exception as e:
+            self.log.error(
+                'Caught an error while trying to switch setups. Setup not switched, '
+                'simulation continues: %s', e)
+            raise
+
+    def _reset_timers(self):
+        self._cycles = 0  # Number of cycles processed
+        self._runtime = 0.0  # Total simulation time processed
```

Comment:
I don't mind much either way, but I'm leaning towards letting it continue counting unless there is a practical reason to reset it.

I can't imagine a scenario where a user would need it to reset to 0 (if nothing else they can save the current cycles/runtime values at the point of changing setups, to use as they new starting point). But I can imagine scenarios where resetting counters can lead to unexpected behaviour (for example, if something is tracking difference in cycle count periodically, and suddenly some other part of the system resets the counters, making the next difference a huge negative number).